src/Compilers/CSharp/Test/Emit2/Semantics/LockTests.cs

```diff
@@ -2038,10 +3344,407 @@ async IAsyncEnumerable<int> M()
                 }
             }
             """;
-        CreateCompilationWithTasksExtensions([source, LockTypeDefinition, AsyncStreamsTypes]).VerifyDiagnostics(
-            // (10,15): error CS9217: A lock statement on a value of type 'System.Threading.Lock' cannot be used in async methods or async lambda expressions.
+        CreateCompilationWithTasksExtensions([source, LockTypeDefinition, AsyncStreamsTypes]).VerifyEmitDiagnostics(
+            // (10,15): error CS4013: Instance of type 'Lock.Scope' cannot be used inside a nested function, query expression, iterator block or async method
```

Comment:
>// (10,15): error CS4013: Instance of type 'Lock.Scope' cannot be used inside a nested function, query expression, iterator block or async method [](http://example.com/codeflow?start=12&length=145)

I think the error is misleading now. The previous and the next tests demonstrate that 'Lock.Scope' can be used inside an iterator block.

Reply:
There's a PROTOTYPE comment in IteratorAndAsyncCaptureWalker to improve this error message.

Reply:
> There's a PROTOTYPE comment in IteratorAndAsyncCaptureWalker to improve this error message.

Consider placing the comment in the test(s), in addition to, or instead.